x-pack/plugins/observability_solution/observability_onboarding/public/application/quickstart_flows/auto_detect/accordion_with_icon.tsx

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { type FunctionComponent } from 'react';
+import {
+  EuiAccordion,
+  EuiIcon,
+  EuiTitle,
+  EuiFlexGroup,
+  EuiFlexItem,
+  type EuiAccordionProps,
+} from '@elastic/eui';
+
+interface AccordionWithIconProps
+  extends Omit<EuiAccordionProps, 'buttonContent' | 'buttonProps' | 'borders' | 'paddingSize'> {
+  title: string;
+  iconType: string;
+}
+export const AccordionWithIcon: FunctionComponent<AccordionWithIconProps> = ({
+  title,
+  iconType,
+  children,
+  ...rest
+}) => {
+  return (
+    <EuiAccordion
+      {...rest}
+      buttonContent={
+        <EuiFlexGroup gutterSize="s" alignItems="center" responsive={false} css={{ marginLeft: 8 }}>
+          <EuiFlexItem grow={false}>
+            <EuiIcon type={iconType} size="l" />
+          </EuiFlexItem>
+          <EuiFlexItem>
+            <EuiTitle size="xs" css={rest.isDisabled ? { color: 'inherit' } : undefined}>
```

Comment:
nit: better to have `isDisabled` as a de-structured prop as you use it directly

Reply:
But then I would also have to manually pass it into accordion. Leaving it inside `rest` means it will get spread with the rest of the props that are just pass through props from the `EuiAccordion`.